### PR TITLE
Fixes and extend json functionality.

### LINKED ISSUE #54: `--json` output seems broken since the 0.1.0 release

### PR DESCRIPTION
**Pull request by jmons**
* Fixes #54,
* introduces a `—json-indent` option that pretty prints

### CODE REVIEW

File: riffdog/command_line.py [hunk 1]
```diff
@@ -161,7 +161,17 @@ def main():
     results = rd._items  # FIXME - accessing internal object
 
     if parsed_args.json:
-        print(dumps(results, cls=ReportEncoder))
+
+        filtered_data = []
+        for item in results:
+            if not item.matched or parsed_args.show_matched:
+                filtered_data.append(item)
```
Comment (dreynolds) [inline riffdog/command_line.py:162]: Is line 165-168 related to this change?

Comment (jmons) [reply riffdog/command_line.py:162]: Good Catch! IT should be - let me just check the next couple of lines

Comment (jmons) [reply riffdog/command_line.py:162]: updated in next commit - these *should* be here, but the next lines should have been using the filtered_data not the original results.

**Review by dreynolds — APPROVED** [review]
Looks good, I've not had a chance to run it though

**Pull request merged**